R/ShinyArchRExports.R

```diff
@@ -0,0 +1,620 @@
+  }
+  
+  # Create coverage objects - should be saved within a dir called ShinyCoverage within the ArchRProjShiny output directory
+  covDir <- file.path(projDir, mainDir, "ShinyCoverage", groupBy)
+  covFiles <- list.files(path = covDir, pattern = "\\_cvg.rds$")
+  
+  # this is still a slightly dangerous comparison, better would be to compare for explicitly the file names that are expected
+  if(length(covFiles) == length(unique(ArchRProjShiny@cellColData[,groupBy]))){
+    if(force){
+      .exportClusterCoverageRDS(ArchRProj = ArchRProjShiny, tileSize = tileSize, groupBy = groupBy, outDir = covDir)
+    } else{
+      message("Coverage files already exist. Skipping fragment file generation...")
+    }
+  }else{
+    dir.create(file.path(mainOutputDir, "ShinyCoverage"))
+    dir.create(covDir, showWarnings = TRUE)
+    .exportClusterCoverageRDS(ArchRProj = ArchRProj, tileSize = tileSize, groupBy = groupBy, fragDir = fragDir, outDir = covDir)
+  }
+  
+  # Create directory to save input data to Shinyapps.io (everything that will be preprocessed)
+  # dir.create(file.path(projDir, mainDir, subOutDir), showWarnings = TRUE)
+  
+  allMatrices <- getAvailableMatrices(ArchRProjShiny)
+  matrices <- list()
+  imputeMatrices <- list()
+  imputeWeights <- getImputeWeights(ArchRProj = ArchRProjShiny)
+  df <- getEmbedding(ArchRProjShiny, embedding = embedding, returnDF = TRUE)
+  
+  if(!file.exists(file.path(projDir,mainDir, subOutDir, "matrices.rds")) && !file.exists(file.path(projDir,mainDir, subOutDir, "imputeMatrices.rds"))){
+    for(matName in allMatrices){
+      if(matName %in% supportedMatrices){
+        
+          featuresNames <- getFeatures(ArchRProj = ArchRProj, useMatrix = matName)
+          dir.create(file.path(projDir,mainDir, subOutDir, matName), showWarnings = FALSE)
+          saveRDS(featuresNames, file.path(projDir,mainDir, subOutDir, matName, paste0(matName, "_names.rds")))
+          
+          if(!is.null(featuresNames)){
+            
+            mat = Matrix(.getMatrixValues(
+              ArchRProj = ArchRProjShiny, 
+              name = featuresNames,
+              matrixName = matName,
+              log2Norm = FALSE,
+              threads = threads), sparse = TRUE)
+            
+            matrices[[matName]] = mat
+            matList = mat[,rownames(df), drop=FALSE]
+            .logThis(matList, paste0(matName,"-Before-Impute"), logFile = logFile)
+            
+            if(getArchRVerbose()) message("Imputing Matrix")
+            imputeMat <- imputeMatrix(mat = as.matrix(matList), imputeWeights = imputeWeights, logFile = logFile)
+            
+            if(!inherits(imputeMat, "matrix")){
+              imputeMat <- mat(imputeMat, ncol = nrow(df))
+              colnames(imputeMat) <- rownames(df)
+            }
+            imputeMatrices[[matName]] <- imputeMat
+            
+      }else{
+            message(matName, " is NULL.")
```

Comment:
when would this condition be triggered? Should this be a `stop()` call instead? Seems like a big problem if you cant create the matrix. Big enough that downstream code would break, making `stop()` the right choice here?

Reply:
[39622f9](https://github.com/paupaiz/ArchR/commit/39622f963602dfa59e38c0a213715c771e0dd480)